pkg/tfgen/docs.go

```diff
@@ -1331,11 +1331,20 @@ func (g *Generator) convertHCLToString(hcl, path, languageName string) (string,
 		// we write an example to a temp file internally in order to pass to convert.Convert().
 		//
 		// fileName starts with a "/" which is not present in the resulting error, so we need to skip the first rune.
-		errMsg := strings.ReplaceAll(diags.All.Error(), fileName[1:], "")
+		errMsg := ""
+		first := true
+		for _, diag := range diags.All {
+			if !first {
+				errMsg += ", and "
+			}
+			errMsg += diag.Error()
+		}
+
+		errMsg = strings.ReplaceAll(errMsg, fileName[1:], "")
```

Comment:
The comment above about removing the temporary filename from the error message should probably be moved here.